### ISSUE TITLE
General resource usage and possible fix

### ISSUE THREAD

**Issue by ircfelix** [stpain/guildbook#42]
Hey there,

I LOVE this addon, but ever since I have started using it, my client got significantly slower. I understand or assume that this comes from live updating the addon, however, it would be nice if you could put some more control to the resource usage of the addon. I think it would be nice, if people could decide when they want to update their guildbook info from the guild, maybe with a very small dialogue or a flashing icon that does not disturb you very much e.g. while raiding. There is nothing worse than having a lag spike from a guildbook update when you log in and need to go somewhere very fast.

Apart from that, please keep up the good work. I'm in a very big guild and we were all told to use guildbook, so this really is an issue!

Have a nice day and a good week and thank you for your efforts coding this gem of an addon!

**Comment by Bonkeybee** [comment]
I agree with this. Guildbook should probably pause or prevent any updating while in combat/dungeon/party/raid/battleground.

Perhaps for some of these scenarios you just throttle the updating to go slower to be less intrusive, or outright prevent it from triggering at all.

A good time to do updates would be when in a rested city or on a flight path.

**Comment by stpain** [comment]
Hmmm, could you explain the issues or a bit more detail.

Guildbook doesnt send any comms when you are in combat or in an instance, this was done to prevent any issues with meter addons (recount, details etc).

On log in the addon will run through a series of requests from the server, these are staggered and for me with about ~1300 takes about 5 mins.

Im curious about when you get lag, so i can look into specific parts, thanks

**Comment by stpain** [comment]
i must apologise i hadnt actually added the "return" to escape comms will add this to next update

**Comment by Bonkeybee** [comment]
Thanks for looking into this

**Comment by stpain** [comment]
ok for now i have added the escapes to all addon comms if in combat or an instance, outside of this i've slowed down the server requests that happen on login.

beyond this i would need to know where issues are happening, thanks
